fonduer/candidates/models/mention.py

```diff
@@ -0,0 +1,183 @@
+import logging
+
+from sqlalchemy import Column, ForeignKey, Integer, String, UniqueConstraint
+from sqlalchemy.orm import backref, relationship
+
+from fonduer.meta import Meta
+from fonduer.utils import camel_to_under
+
+_meta = Meta.init()
+logger = logging.getLogger(__name__)
+
+
+class Mention(_meta.Base):
+    """
+    An abstract Mention.
+
+    New mention types should be defined by calling mention_subclass(),
+    **not** subclassing this class directly.
+    """
+
+    __tablename__ = "mention"
+    id = Column(Integer, primary_key=True)
+    type = Column(String, nullable=False)
+    split = Column(Integer, nullable=False, default=0, index=True)
+
+    __mapper_args__ = {"polymorphic_identity": "mention", "polymorphic_on": type}
+
+    def get_contexts(self):
+        """Get the consituent context making up this mention."""
+        return tuple(getattr(self, name) for name in self.__argnames__)
+
+    def get_cids(self):
+        """Get the canonical IDs (CIDs) of the context making up this mention."""
+        return tuple(getattr(self, name + "_cid") for name in self.__argnames__)
+
+    def __len__(self):
+        return len(self.__argnames__)
+
+    def __getitem__(self, key):
+        return self.get_contexts()[key]
+
+    def __repr__(self):
+        return "%s(%s)" % (
+            self.__class__.__name__,
+            ", ".join(map(str, self.get_contexts())),
+        )
+
+    def __gt__(self, other):
+        # Allow sorting by comparing the string representations of each
+        return self.__repr__() > other.__repr__()
+
+
+# This global dictionary contains all classes that have been declared in this
+# Python environment, so that mention_subclass() can return a class if it
+# already exists and is identical in specification to the requested class
+mention_subclasses = {}
```

Comment:
Why is this here? It should be somehwere high level?

Reply:
Will move up in the file.